00_Base/src/interfaces/repository.ts

```diff
@@ -3,21 +3,66 @@
 //
 // SPDX-License-Identifier: Apache 2.0
 
+import { EventEmitter } from 'events';
+
+export type CrudEvent<T> = {
```

Comment:
should CrudEvent be a generic class?

Reply:
this way we can have builder(s), helper methods, (de)serializer(s), annotation(s) etc... and should be more stable I think in general with classes since these represent data objects

Reply:
CrudEvent is used exclusively to provide typing for the supported events.
In a code block such as below, your IDE will only permit the events in CrudEvent to be used as values for eventName, and it will automatically provide the typing for the callback argument.

```
const variableStatus = new SequelizeRepository<VariableStatus>(config, VariableStatus.MODEL_NAME, logger, sequelizeInstance);
const eventName = 'created';
variableStatus.on(eventName, async (variableStatuses) => { 
    // typeof  variableStatuses = VariableStatus[]
});
```

I.E. if you changed eventName to 'create' it would not compile, and if you are preventing implicit 'any' typing this will still lint okay since the callback argument (variableStatuses) is explicitly typed.
This works because it is generic, since it can utilize the same generic typing as CrudRepository implementations when used in the function signatures of on() and emit().